### DISCUSSION TITLE
Inconsistent restrictions on create/edit for M2M relation table

### DISCUSSION THREAD

**Discussion by henk23** [directus/directus#17502] [Archive]
I have a M2M relation table `events_files` between `events` and `directus_files`.

When creating an event, I can set 

```
{
  images: {
    create: [
      {
        directus_files_id: {id: 'uuid of file'},
        events_id: '+'
      }
    ]
  }
}
```

and the relationship is created correctly, although the user does **not** have any 'create' permissions on `events_files`. 
When I try to update this event, using the exact same syntax, I get the Error 403 FORBIDDEN.

So to fix this issue, I tried to create a custom permission for `events_files` 'create' action, but it won't work.
For 'read', 'update' and 'delete' actions, I have this custom item permission set and it works:

```
{
  "events_id": {
    "organized_by": {
      "_eq": "$CURRENT_USER"
    }
  }
}
```

The App won't let me set this permission for the 'create' action. And when I put it in as a raw value, when trying to update the event, it gives me this error:

```events_id: validationError.undefined```

Sooooo, to summarize, I see 3 issues here:

1. Should the user be able to create a row in `events_files` (via creating an event) although they have no permissions set for this?
2. Should the permissions not be the same as when updating events?
3. Why can't I set a relational permission to 'create', the same as I can to 'read', 'update' and 'delete'?

**Comment by rijkvanzanten** [comment]
> 1. Should the user be able to create a row in events_files (via creating an event) although they have no permissions set for this?

No, if the user doesn't have permissions to create items in a collection, they shouldn't be allowed to create items in said collection. 

> 2. Should the permissions not be the same as when updating events?

It depends on what you're doing through the parent item. If you're updating the `events_files` row through the parent, the update permissions of `events_files` should be used. The junction table is just another collection, and the nested modification syntax is just a shorthand of doing two calls (one to modify the parent, and one to modify the child).

> 3. Why can't I set a relational permission to 'create', the same as I can to 'read', 'update' and 'delete'?

I think you're confusing "Item Permissions" with "Fields". "Fields" controls what fields you're able to modify, whereas "Item Permissions" allows you to control _which_ item you can modify. For creations, there is no item to read/modify yet, so there's no "Item Permissions". The Fields you can read/update aren't relational either

**Reply by henk23** [reply]
Hey @rijkvanzanten, thanks for the quick reply.

> No, if the user doesn't have permissions to create items in a collection, they shouldn't be allowed to create items in said collection.
> 
> > 2. Should the permissions not be the same as when updating events?
> 
> It depends on what you're doing through the parent item. If you're updating the `events_files` row through the parent, the update permissions of `events_files` should be used. The junction table is just another collection, and the nested modification syntax is just a shorthand of doing two calls (one to modify the parent, and one to modify the child).

Thanks for clarifying. Then there's a bug here, or even a security vulnerability. Because when initially creating an `event` with `images.create` as described above, a row in `events_files` is created, although the user has no 'create' permission. When editing the event later (trying to create a new relationship), the permission is correclty denied.

> > 3. Why can't I set a relational permission to 'create', the same as I can to 'read', 'update' and 'delete'?
> 
> I think you're confusing "Item Permissions" with "Fields". "Fields" controls what fields you're able to modify, whereas "Item Permissions" allows you to control _which_ item you can modify. For creations, there is no item to read/modify yet, so there's no "Item Permissions". The Fields you can read/update aren't relational either

Sorry, yes, I confused the terms, I meant "Field Validation". What I want to do is this: The user should only be allowed to create rows in `events_files` for events that they are the organizer of (`event.organized_by === $CURRENT_USER`). This seems not to be possible at the moment, correct me if I'm wrong.
